core/controllers/reader.py

```diff
@@ -1005,15 +1005,25 @@ def get(self):
 
         skill_ids = self.request.get('skill_ids').split(',')
         question_count = self.request.get('question_count')
+        fetch_by_mastery = self.request.get('fetch_by_mastery')
 
         if not question_count.isdigit() or int(question_count) <= 0:
             raise self.InvalidInputException(
                 'Question count has to be greater than 0')
 
+        if not (fetch_by_mastery == 'true' or fetch_by_mastery == 'false'):
+            raise self.InvalidInputException('fetch_by_mastery must be 0 or 1')
+        if fetch_by_mastery == 'true':
+            fetch_by_mastery = True
```

Comment:
I don't think you should use the same variable here. See my above comment about renaming the string variable. Also you can replace this whole block with:
```
fetch_by_mastery = (fetch_by_mastery_value == 'true')
```

Reply:
Done.